app/scripts/modules/core/src/pipeline/config/triggers/artifacts/artifact.component.ts

```diff
@@ -1,20 +1,84 @@
-import { IComponentController, IComponentOptions, module } from 'angular';
+import {
+  ICompileService,
+  IComponentOptions,
+  IController,
+  IControllerService,
+  IRootElementService,
+  IRootScopeService,
+  module
+} from 'angular';
+import { IArtifact, IArtifactKindConfig } from 'core/domain';
+import { PipelineConfigProvider } from 'core/pipeline';
 
-import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
-import { IArtifact } from 'core/domain/IArtifact';
-
-class ArtifactController implements IComponentController {
+class ArtifactCtrl implements IController {
   public artifact: IArtifact;
+  public options: IArtifactKindConfig[];
+  public description: string;
+
+  constructor(private pipelineConfig: PipelineConfigProvider,
+              private $controller: IControllerService,
+              private $compile: ICompileService,
+              private $element: IRootElementService,
+              private $rootScope: IRootScopeService) {
+    'nginject';
+    this.options = this.pipelineConfig.getArtifactKinds();
+    this.loadArtifactKind();
+  }
+
+  public loadArtifactKind(): void  {
+    const kind = this.artifact.kind;
+    if (!kind) {
+      return;
+    }
+    const triggerConfig = this.options.filter(function(config) {
+      return config.key === kind;
+    });
+
+    if (triggerConfig.length) {
+      const config = triggerConfig[0];
+      const template: Element = config.template as any;
+      this.description = config.description;
+
+      const ctrl = config.controller;
+      const controller = this.$controller(ctrl, {artifact: this.artifact});
+      (controller as any).onInit();
+      const scope = this.$rootScope.$new();
+      scope[config.controllerAs] = controller;
+
+      const templateBody = this.$compile(template)(scope) as any;
+      this.$element.find('.artifact-body').html(templateBody);
+    }
+  }
 }
 
 class ArtifactComponent implements IComponentOptions {
-  public bindings = { artifact: '=' };
-  public templateUrl = require('./artifact.html');
-  public controller = ArtifactController;
+  public bindings: any = {artifact: '='};
+  public controller: any = ArtifactCtrl;
   public controllerAs = 'ctrl';
+  public template = `
+<div class="form-group">
+  <div class="col-md-2 col-md-offset-1">
+    <select class="input-sm"
+            required
+            ng-change="ctrl.loadArtifactKind()"
+            ng-options="option.key as option.label for option in ctrl.options"
+            ng-model="ctrl.artifact.kind">
+      <option style="display:none" value="">Select a kind</option>
```

Comment:
Yup, but only when `ctrl.artifact.kind` is null (if someone manually edits the JSON). Otherwise it's not an option, and disappears when something is selected.